.github/workflows/pretest.yml

```diff
@@ -50,6 +50,11 @@ jobs:
         autopep8 -r . --diff | tee check_autopep8
         test ! -s check_autopep8
 
+    - name: Static Type Checking (mypy)
+      run: |
+        pip install numpy==1.21.* mypy==0.790
```

Comment:
* Are there any reasons to use old versions of numpy/mypy here?
* How about adding `mypy` to `setup.py` (`test`) so that contributors can use the same mypy version as CI?